atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1773,6 +1774,27 @@ private void alterGcAndTombstoneWithLock(
         }
     }
 
+    public void cleanUpSchemaMutationLockTablesState() throws Exception {
+        // get a unique table and create it if it doesn't exist
+        Set<TableReference> tables = lockTables.getAllLockTables();
+        TableReference unique;
+        if (tables.size() == 0) {
+            unique = lockTables.createLockTable();
+            log.info("There were no locks tables, so one was succesfully created: {}", unique.toString());
+            return;
+        }
+        unique = tables.iterator().next();
+        tables.remove(unique);
+
+        // drop additional unwanted tables
+        if (tables.size() > 0) {
+            dropTablesWithLock(tables);
```

Comment:
Also, nit: I think `dropTablesWithLock` (as well as `createTablesWithLock` et al) are poorly named because they don't actually grab any locks. Maybe worth renaming to `dropTablesInternal` (or perhaps you can think of a better name)